src/sentry/integrations/slack/requests.py

```diff
@@ -92,7 +96,22 @@ def _validate_data(self):
             raise SlackRequestError(status=400)
 
     def _authorize(self):
-        if self.data.get("token") != options.get("slack.verification-token"):
+        signing_secret = options.get("slack.signing-secret")
+        # use the signing_secret if it's available
+        if signing_secret:
+            # Taken from: https://github.com/slackapi/python-slack-events-api/blob/master/slackeventsapi/server.py#L47
+            # Slack docs on this here: https://api.slack.com/authentication/verifying-requests-from-slack#about
+            signature = self.request.META["HTTP_X_SLACK_SIGNATURE"]
```

Comment:
is it possible for the signature to be missing? would we need to handle that?

Reply:
@MeredithAnya Good question. I don't think that should happen, but if it does, it's going to blow up with a `KeyError` and return a `500`. I think that's probably what we want since that error will go to Sentry and it will be clear what the problem is.  I think that's probably better than just defaulting to an empty string and returning a `401` which is how it's done in the code snippet I linked to.